kafka/metrics_test.go

```diff
@@ -557,6 +557,9 @@ func setupTestProducer(t testing.TB, tafunc TopicAttributeFunc) (*Producer, sdkm
 			MeterProvider:      mp,
 			TopicAttributeFunc: tafunc,
 		},
+		ProduceCallback: func(*kgo.Record, error) {
+			time.Sleep(1 * time.Millisecond)
```

Comment:
I initially did have a `<-time.After(1*time.Millisecond)` between the `Produce` and `Close`. Didn't really know which "spot" to place it is the best.

The close should not affect the records being produced on the queue, since `Produce` is waiting properly. Even `Close` if blocking with the `Flush` method being called. 

Also, I do not understand why it is only the `producer.messages` metrics that fail. Is there any links on how OTEL handles concurrency? Reading the `Collect` code is actually pretty tricky also. 

I'll implement you channel concept. Good idea.